dimod/generators/mimo.py

```diff
@@ -0,0 +1,1085 @@
+            When ``False``, spin variables are labeled with ``geometric_position`` 
+            index at geometric position. This option is currently not implemented. 
+
+        transmitted_symbols: Set of symbols transmitted. Used in combination 
+            with ``F`` to generate the received signal, :math:`y`. The number 
+            of transmitted symbols must be consistent with ``F``.
+            
+            For BPSK and QPSK modulations, statistics of the ensemble do not 
+            depend on the choice: all choices are equivalent. By default, 
+            symbols are chosen for all users as :math:`1` or :math:`1 + 1j`, 
+            respectively. Note that for correct analysis by some solvers, 
+            applying spin-reversal transforms may be necessary.
+            
+            For QAM modulations, amplitude randomness affects likelihood in a 
+            non-trivial way. By default, symbols are chosen independently and 
+            identically distributed from the constellations. 
+        
+        channel_noise: Channel noise as a complex value.
+
+        num_transmitters_per_node: Number of users. Each user transmits one 
+            symbol per frame. Overrides any ``num_transmitters`` attribute of
+            a :class:`networkx.Graph` provided as the ``lattice`` parameter. 
+
+        num_receivers_per_node: Number of receivers of a channel. Must be 
+            consistent with the length of any provided signal, ``len(y)``.
+            Overrides any ``num_receivers`` attribute of the ``lattice`` 
+            parameter.
+
+        SNRb: Signal-to-noise ratio per bit, :math:`SNRb=10^{SNR_b[decibels]/10}`, 
+            used to generate the noisy signal when ``y`` is not provided. 
+            If ``float('Inf')``, no noise is added.  
+        
+        seed: Random seed, as an integer, or state, as a 
+            :class:`numpy.random.RandomState` instance.
+        
+        F_distribution: Zero-mean, variance-one distribution, in tuple form 
+            ``(distribution, type)``, used to generate each element in ``F`` 
+            when ``F`` is not provided . Supported values are:
+            
+            * ``'normal'`` or ``'binary'`` for the distribution 
+            * ``'real'`` or ``'complex'`` for the type 
+            
+            For large numbers of receivers and transmitters, statistical 
+            properties of the likelihood are weakly dependent on the 
+            distribution. Choosing ``'binary'`` allows for integer-valued 
+            Hamiltonians while ``'normal'`` is a more typical model. The channel 
+            can be real or complex; in many cases this represents a superficial 
+            distinction up to rescaling. For real-valued symbols (BPSK) the 
+            default is ``('normal', 'real')``; otherwise, the default is 
+            ``('normal', 'complex')``.
+        
+        use_offset: Adds a constant to the Ising model energy so that the 
```

Comment:
Why not always include the offset? It's hard for me to imagine what harm it would do. And if the user really doesn't want it they can just
```python
bqm.offset = 0
```